lib/activerecord-mysql-unsigned/active_record/v3/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -37,12 +38,27 @@ def native
         end
       end
 
+      class Column < ConnectionAdapters::Column # :nodoc:
+        attr_accessor :extra
+      end
+
+      def columns(table_name, name = nil) #:nodoc:
+        sql = "SHOW FULL FIELDS FROM #{quote_table_name(table_name)}"
+        execute_and_free(sql, 'SCHEMA') do |result|
+          each_hash(result).map do |field|
+            new_column(field[:Field], field[:Default], field[:Type], field[:Null] == "YES", field[:Collation]).tap do |column|
+              column.extra = field[:Extra]
+            end
+          end
+        end
+      end
+
       def schema_creation
         SchemaCreation.new self
       end
 
       def create_table_definition(name, temporary, options)
-        TableDefinition.new self
+        PseudoTableDefinition.new self
```

Comment:
The name was ambiguous, so renamed `TableDefinition` to `PseudoTableDefinition`.